Core/MIPS/MIPSVFPUUtils.cpp

```diff
@@ -277,16 +308,49 @@ void WriteMatrix(const float *rd, MatrixSize size, int reg) {
 		ERROR_LOG_REPORT(CPU, "Write mask used with vfpu matrix instruction.");
 	}
 
-	for (int i = 0; i < side; i++) {
-		for (int j = 0; j < side; j++) {
-			// Hm, I wonder if this should affect matrices at all.
-			if (j != side -1 || !currentMIPS->VfpuWriteMask(i))	{
-				int index = mtx * 4;
-				if (transpose)
-					index += ((row+i)&3) + ((col+j)&3)*32;
-				else
-					index += ((col+j)&3) + ((row+i)&3)*32;
-				V(index) = rd[j*4+i];
+	// The voffset ordering is now integrated in these formulas,
+	// eliminating a table lookup.
+	float *v = currentMIPS->v + (size_t)mtx * 16;
+	if (transpose) {
+		// Fast path
+		if (side == 4 && row == 0 && col == 0 && currentMIPS->VfpuWriteMask() == 0x0) {
+			for (int j = 0; j < side; j++) {
+				for (int i = 0; i < side; i++) {
+					int index = i * 4 + j;
+					v[index] = rd[j * 4 + i];
+				}
+			}
+		} else {
+			for (int j = 0; j < side; j++) {
+				for (int i = 0; i < side; i++) {
+					// Hm, I wonder if this should affect matrices at all.
```

Comment:
We can possibly remove this comment now - in most cases it affects the last row or the last register in the matrix.  I think interpreter handles all the cases by setting the write mask appropriately.

-[Unknown]